src/controllers/dynakube/version/reconciler.go

```diff
@@ -84,12 +81,14 @@ func (reconciler *Reconciler) createDockerConfigWithCustomCAs(ctx context.Contex
 }
 
 func (reconciler *Reconciler) needsReconcile(updaters []versionStatusUpdater) bool {
+	neededUpdaters := []versionStatusUpdater{}
 	for _, updater := range updaters {
 		if reconciler.needsUpdate(updater) {
-			return true
+			neededUpdaters = append(neededUpdaters, updater)
 		}
 	}
-	return false
+	updaters = neededUpdaters
```

Comment:
can you just return `neededUpdaters` and assign it to `updaters` outside this method? currently this has "hidden" side effects